### PR TITLE
Fix SEGFAULT in test_fstring

### PR DESCRIPTION
**Pull request by lysnikolaou**
In `fstring_compile_expr` the parser was being invoked by calling
`parse` directly, instead of `run_parser`. Thus, the jmp_buf wasn't
being initialized, leading to an invalid jump, upon calling `longjmp`.

Turns out, this was there all before #57, but came up due to calls to
`longjmp` there.

### CODE REVIEW

File: Parser/pegen/pegen.c [hunk 1]
```diff
@@ -703,7 +703,7 @@ Parser_New(struct tok_state *tok, int start_rule, int *errcode, PyArena *arena)
     return p;
 }
 
-mod_ty
```
Comment (pablogsal) [inline Parser/pegen/pegen.c:706]: Why the change of type? Isn't `parse` returning always a `mod_ty`?

Comment (lysnikolaou) [reply Parser/pegen/pegen.c:706]: > Why the change of type? Isn't parse returning always a mod_ty

No, in case `p->start_rule` is `Py_fstring_input`, `parse` calls `fstring_rule`, which directly returns an `expr_ty`.

> if not, we should update as well the type of `res`

Done!

**Comment by pablogsal** [comment]
Great catch! 🎣

File: Parser/pegen/parse_string.c [hunk 2]
```diff
@@ -588,10 +588,9 @@ fstring_compile_expr(Parser *p, const char *expr_start, const char *expr_end,
 
     Parser *p2 = Parser_New(tok, Py_fstring_input, NULL, p->arena);
 
-    expr = parse(p2);
+    expr = run_parser(p2);
 
     if (expr == NULL){
```
Comment (gvanrossum) [inline Parser/pegen/parse_string.c:593]: Pep7: space before {.

File: Parser/pegen/pegen.c (hunk 1 above)
Comment (gvanrossum) [reply Parser/pegen/pegen.c:706]: Every void* makes me sad... But I understand.
